homeassistant/components/flexit_bacnet/__init__.py

```diff
@@ -0,0 +1,49 @@
+"""The Flexit Nordic (BACnet) integration."""
+from __future__ import annotations
+
+import asyncio.exceptions
+import logging
+
+from flexit_bacnet import FlexitBACnet
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [Platform.CLIMATE]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Flexit Nordic (BACnet) from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    device = FlexitBACnet(entry.data["address"], entry.data["device_id"])
+
+    await device.update()
+    try:
+        await device.update()
+    except asyncio.exceptions.TimeoutError:
+        _LOGGER.error("Cannot connect to Flexit Nordic unit")
+        return False
+
+    if not device.is_valid():
```

Comment:
@frenck it means that a BACnet peer didn't return valid information (in this case it checks for a serial number of the device).
It probably had more meaning in the synchronous version of `flexit_bacnet` library, but now that I think about it, in the v2 of the library it might be redundant. I'll remove it.